apps/src/craft/designer/craft.js

```diff
@@ -0,0 +1,913 @@
+    contentDiv: popupDiv,
+    defaultBtnSelector: '#choose-steve',
+    onHidden: function () {
+      onSelectedCallback(selectedPlayer);
+    },
+    id: 'craft-popup-player-selection',
+  });
+  dom.addClickTouchEvent($('#close-character-select')[0], function () {
+    popupDialog.hide();
+  }.bind(this));
+  dom.addClickTouchEvent($('#choose-steve')[0], function () {
+    selectedPlayer = CHARACTER_STEVE;
+    trackEvent('MinecraftDesigner', 'ClickedCharacter', selectedPlayer);
+    popupDialog.hide();
+  }.bind(this));
+  dom.addClickTouchEvent($('#choose-alex')[0], function () {
+    selectedPlayer = CHARACTER_ALEX;
+    trackEvent('MinecraftDesigner', 'ClickedCharacter', selectedPlayer);
+    popupDialog.hide();
+  }.bind(this));
+  popupDialog.show();
+};
+
+Craft.clearPlayerState = function () {
+  window.localStorage.removeItem('craftSelectedPlayer');
+};
+
+Craft.initializeAppLevel = function (levelConfig) {
+  Craft.foldInEntities(levelConfig);
+
+  var fluffPlane = [];
+  // TODO(bjordan): remove configuration requirement in visualization
+  for (var i = 0; i < (levelConfig.gridWidth || 10) * (levelConfig.gridHeight || 10); i++) {
+    fluffPlane.push('');
+  }
+
+  var levelAssetPacks = {
+    beforeLoad: Craft.minAssetsForLevelWithCharacter(levelConfig.puzzle_number),
+    afterLoad: Craft.afterLoadAssetsForLevel(levelConfig.puzzle_number)
+  };
+
+  const doNotShowPlayer = levelConfig.usePlayer === false;
+
+  Craft.gameController.loadLevel({
+    onDayCallback: () => {
+      Craft.musicController.setGroup('day');
+      Craft.musicController.fadeOut();
+    },
+    onNightCallback: () => {
+      Craft.musicController.setGroup('night');
+      Craft.musicController.fadeOut();
+    },
+    levelVerificationTimeout: levelConfig.levelVerificationTimeout,
+    isDaytime: levelConfig.isDaytime,
+    groundPlane: levelConfig.groundPlane,
+    usePlayer: !doNotShowPlayer,
+    useScore: levelConfig.useScore,
+    entities: levelConfig.entities,
+    isEventLevel: levelConfig.isEventLevel,
+    groundDecorationPlane: levelConfig.groundDecorationPlane,
+    actionPlane: levelConfig.actionPlane,
+    fluffPlane: fluffPlane,
+    playerStartPosition: levelConfig.playerStartPosition,
+    playerStartDirection: levelConfig.playerStartDirection,
+    playerName: Craft.getCurrentCharacter(),
+    assetPacks: levelAssetPacks,
+    gridDimensions: levelConfig.gridWidth && levelConfig.gridHeight ?
+        [levelConfig.gridWidth, levelConfig.gridHeight] :
+        null,
+    // eslint-disable-next-line no-eval
+    verificationFunction: eval('[' + levelConfig.verificationFunction + ']')[0], // TODO(bjordan): add to utils
+    // eslint-disable-next-line no-eval
+    failureCheckFunction: eval('[' + levelConfig.failureCheckFunction + ']')[0], // TODO(bjordan): add to utils
+    // eslint-disable-next-line no-eval
+    timeoutResult: eval('[' + (levelConfig.timeoutVerificationFunction || `function() { return false; }`) + ']')[0],
```

Comment:
Agree, let's move them out of the .level files once the progression levels are locked down and before UI/level tests get added.